### PR TITLE
[Memgraph-2.19 < Helm chart updates] Add configs and notes

### PR DESCRIPTION
**Pull request by antejavor**
### Description

Updates made on the helm charts that will be available with 2.19. 

### Pull request type

Please check what kind of PR this is:

- [x] Fix or improvement of an existing page
- [ ] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
Multiple updates to Helm charts:  https://github.com/memgraph/helm-charts/milestone/5

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [ ] If release-related, add a product and version label
- [ ] If release-related, add release note on product PR

### CODE REVIEW

**Review by kgolubic — APPROVED** [review]
Everything looks OK to me.

File: pages/getting-started/install-memgraph/kubernetes.mdx [hunk 1]
```diff
@@ -91,6 +91,8 @@ their default values.
 | `replicaCount`                              | Number of Memgraph instances to run. Note: no replication or HA support.                            | `1`                                     |
 | `affinity.nodeKey`                          | Key for node affinity (Preferred)                                                                   | `""`                                    |
 | `affinity.nodeValue`                        | Value for node affinity (Preferred)                                                                 | `""`                                    |
+| `nodeSelector`                              | Node selectors for pod. Left empty by default.                                                      | `{}`                                    |
+| `tolerations`                               | Tolerations for pod. Left empty by default.                                                         | `[]`                                    |
```
Comment (hal-eisen-MG) [inline pages/getting-started/install-memgraph/kubernetes.mdx:95]: Avoid defining a term with itself. Needs more explanation.

File: pages/getting-started/install-memgraph/kubernetes.mdx [hunk 2]
```diff
@@ -91,6 +91,8 @@ their default values.
 | `replicaCount`                              | Number of Memgraph instances to run. Note: no replication or HA support.                            | `1`                                     |
 | `affinity.nodeKey`                          | Key for node affinity (Preferred)                                                                   | `""`                                    |
 | `affinity.nodeValue`                        | Value for node affinity (Preferred)                                                                 | `""`                                    |
+| `nodeSelector`                              | Node selectors for pod. Left empty by default.                                                      | `{}`                                    |
```
Comment (hal-eisen-MG) [inline pages/getting-started/install-memgraph/kubernetes.mdx:94]: Avoid defining a term with itself. Needs more explanation.

File: pages/getting-started/install-memgraph/kubernetes.mdx [hunk 3]
```diff
@@ -111,8 +113,10 @@ their default values.
...
-| `memgraphUser`                              | User for the Memgraph database                                                                      | `""`                                    |
-| `memgraphPassword`                          | Password for the Memgraph database                                                                  | `""`                                    |
+| `secrets.enabled`                           | Enable the use of Kubernetes secrets for Memgraph credentials                                       | `false`                                 |
+| `secrets.name`                              | The name of the Kubernetes secret containing Memgraph credentials                                   | `memgraph-secrets`                      |
+| `secrets.userKey`                           | The key in the Kubernetes secret for the Memgraph user                                              | `MEMGRAPH_USER`                         |
+| `secrets.passwordKey`                       | The key in the Kubernetes secret for the Memgraph password                                          | `MEMGRAPH_PASSWORD`                     |
```
Comment (hal-eisen-MG) [inline pages/getting-started/install-memgraph/kubernetes.mdx:119]: What is a "key" in this context? Why is the string "key" present in the parameter name, but missing in the environment variable - that's confusing and inconsistent.

Comment (antejavor) [reply pages/getting-started/install-memgraph/kubernetes.mdx:119]: Here is the problem: this is not an `env` variable, this is a secret key, that you go and set some value to, and then that value is passed to the actuall env. 

Like this: 
```
            - name: MEMGRAPH_USER
              valueFrom:
                secretKeyRef:
                  name: {{ .Values.secrets.name }}
                  key: {{ .Values.secrets.userKey }}
            - name: MEMGRAPH_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ .Values.secrets.name }}
                  key: {{ .Values.secrets.passwordKey }}
```

You set the key values like this: 
```kubectl create secret generic memgraph-secrets --from-literal=MEMGRAPH_USER=myuser --from-literal=MEMGRAPH_PASSWORD=mypassword```

So the value is implicitly read from the key and set to the `MEMGRAPH_USER` and `MEMGRAPH_PASSWORD`. In this case, the key share the same naming as the env variables. But I can agree it is a bit confusing. 

I will rename this to a USER and PASSWORD, so there is a distinction form envs, but you still have the key step.

**Review by hal-eisen-MG — APPROVED** [review]
LGTM

**Pull request merged**